apps/playground/src/portable-text-toolbar.tsx

```diff
@@ -105,7 +106,7 @@ export function PortableTextToolbar(props: {
 
 function StyleSelector(props: {
   schemaDefinition: SchemaDefinition
-  editor: PortableTextEditor
+  editor: PortableTextEditorInstance
```

Comment:
This PR looks promising, but I think it's a breaking change (both in PTE but also for Studio users) that you can't reference the `PortableTextEditor` as a type anymore.

In any case, this PR gave me an idea for a different take on how to tackle this problem. I'll try to open another PR to compare with.

Reply:
It's always been a nasty anti pattern given that it's a react class component. If you can get it working by reducing the class component to a very thin layer without logic (and preferably use PureComponent instead of Component since the React Compiler only supports function components and will never add class components) then that could work I suppose 🤔